lib/StripeResource.js

```diff
@@ -228,21 +228,31 @@ StripeResource.prototype = {
       return true;
     }
 
-    // Retry on conflict and availability errors.
-    if (res.statusCode === 409 || res.statusCode === 503) {
+    if (res.headers && res.headers['stripe-should-retry'] === 'false') {
+      return false;
+    }
+    if (res.headers && res.headers['stripe-should-retry'] === 'true') {
+      return true;
+    }
+
+    // Retry on conflict errors.
+    if (res.statusCode === 409) {
       return true;
     }
 
-    // Retry on 5xx's, except POST's, which our idempotency framework
-    // would just replay as 500's again anyway.
-    if (res.statusCode >= 500 && res.req._requestEvent.method !== 'POST') {
+    // Retry on 500, 503, and other internal errors.
+    //
+    // Note that we expect the stripe-should-retry header to be false
+    // in most cases when a 500 is returned, since our idempotency framework
+    // would typically replay it anyway.
+    if (res.statusCode >= 500) {
       return true;
     }
 
     return false;
   },
 
-  _getSleepTimeInMS(numRetries) {
+  _getSleepTimeInMS(numRetries, retryAfter) {
```

Comment:
Is it worth making sure, at some point, that `retryAfter` is actually a (reasonably-sized) positive integer?

Reply:
Since we're doing a `Math.max` on this value a few lines later, I don't think we need to explicitly check this value. We should however set a default value here `retryAfter = 0`

Reply:
Eh, I think this was a good point in retrospect, and also helped resolve the `|| 0` debate below. Added an integer check and <60, since we should definitely never ask for something longer than that.